vars/functionalTest.groovy

```diff
@@ -108,15 +108,29 @@ Map call(Map config = [:]) {
 
     sh label: 'Install Launchable',
      script: '''# can't use --upgrade with pip3 because it tries to upgrade everything
```

Comment:
I wonder if this does what you want?
```
--user --upgrade --upgrade-strategy only-if-needed launchable
```

Reply:
It appears to:

```
$ pip3 install --upgrade --upgrade-strategy only-if-needed launchable
Defaulting to user installation because normal site-packages is not writeable
Requirement already satisfied: launchable in /home/brian/.local/lib/python3.12/site-packages (1.90.1)
Collecting launchable
  Downloading launchable-1.90.2-py3-none-any.whl.metadata (2.5 kB)
Requirement already satisfied: urllib3>=1.26 in /usr/lib/python3.12/site-packages (from launchable) (1.26.19)
Requirement already satisfied: junitparser>=2.0.0 in /home/brian/.local/lib/python3.12/site-packages (from launchable) (3.1.2)
Requirement already satisfied: setuptools in /usr/lib/python3.12/site-packages (from launchable) (69.0.3)
Requirement already satisfied: python-dateutil in /usr/lib/python3.12/site-packages (from launchable) (2.8.2)
Requirement already satisfied: tabulate in /usr/lib/python3.12/site-packages (from launchable) (0.9.0)
Requirement already satisfied: click>=8.1 in /usr/lib/python3.12/site-packages (from launchable) (8.1.7)
Requirement already satisfied: requests>=2.25 in /usr/lib/python3.12/site-packages (from launchable) (2.31.0)
Requirement already satisfied: more-itertools>=7.1.0 in /usr/lib/python3.12/site-packages (from launchable) (10.1.0)
Requirement already satisfied: charset-normalizer<4,>=2 in /usr/lib/python3.12/site-packages (from requests>=2.25->launchable) (3.3.2)
Requirement already satisfied: idna<4,>=2.5 in /usr/lib/python3.12/site-packages (from requests>=2.25->launchable) (3.7)
Requirement already satisfied: six>=1.5 in /usr/lib/python3.12/site-packages (from python-dateutil->launchable) (1.16.0)
Downloading launchable-1.90.2-py3-none-any.whl (9.4 MB)
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 9.4/9.4 MB 5.0 MB/s eta 0:00:00
Installing collected packages: launchable
  Attempting uninstall: launchable
    Found existing installation: launchable 1.90.1
    Uninstalling launchable-1.90.1:
      Successfully uninstalled launchable-1.90.1
Successfully installed launchable-1.90.2
```

Although I cannot be sure what side-effects it might have, if any.

Even though I have a solution that seems to be working already, I will push a new one with your suggestion.